cmd/vsphere-xcopy-volume-populator/e2e-tests/Dockerfile

```
@@ -0,0 +1,84 @@
+# Define versions and checksums for security and reproducibility
+ENV OC_CLIENT_VERSION=4.14.8
+ENV OC_CLIENT_SHA256=2c992ec3beecc9bf305a2a35035262afe515de2661737cc5e14d112ee1d9c7c6
+ENV ANSIBLE_VERSION=9.5.1
+ENV PYVMOMI_VERSION=8.0.3.0.1
+
+# Set up and activate a virtual environment to safely install Python packages
+ENV VIRTUAL_ENV=/opt/venv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+# Install dependencies, including Ansible, oc, and kubectl
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
```

Hadolint (2.12.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)